### PR TITLE
overlord/state: Add State() methods to state.{Task,Change}

### PR DESCRIPTION
**Pull request by mvo5**
Trivial helper branch for the StateManager branch.

### CODE REVIEW

File: overlord/state/change.go [hunk 1]
```diff
@@ -175,6 +175,11 @@ func (c *Change) SetStatus(s Status) {
 	c.status = s
 }
 
+// State return the pointer to the global State
```
Comment (zyga) [inline overlord/state/change.go]: It's not really global. I'd word this differently. Perhaps ``change specific State``

File: overlord/state/task.go [hunk 2]
```diff
@@ -124,6 +124,11 @@ func (t *Task) SetStatus(s Status) {
 	t.status = s
 }
 
+// State return the pointer to the global State
```
Comment (zyga) [inline overlord/state/task.go]: Same comment as above.

File: overlord/state/change.go (hunk 1 above)
Comment (mvo5) [inline overlord/state/change.go]: Good point, maybe `State returns the pointer to the system State` ?

**Comment by chipaca** [comment]
:+1:

File: overlord/state/change.go [hunk 3]
```diff
@@ -175,6 +175,11 @@ func (c *Change) SetStatus(s Status) {
 	c.status = s
 }
 
+// State returns the pointer to the system State
```
Comment (pedronis) [inline overlord/state/change.go]: why "pointer"?  I don't remember seeing that used so explicitly in the doc comment

**Comment by pedronis** [comment]
looks good, wondering about the doc comments though

File: overlord/state/task.go [hunk 4]
```diff
@@ -124,6 +124,11 @@ func (t *Task) SetStatus(s Status) {
 	t.status = s
 }
 
+// State returns the pointer to the system State
```
Comment (pedronis) [inline overlord/state/task.go]: this needs "the pointer to" removed as well

Comment (mvo5) [inline overlord/state/task.go]: Good catch and sorry that I failed to spot it in the previous commit. Fixed now :)

**Pull request merged**